commit aa786177ee5b7c5f1acd39ee19f682957b17319f
Author: Daniel Cordero <gentoo.iproute2@0xdc.xyz>
Date:   Sun Jul 28 22:05:49 2024 +0000

    Add magic libtirpc flag handling

diff --git a/config.include b/config.include
index 19db3b45..999e77fa 100644
--- a/config.include
+++ b/config.include
@@ -18,6 +18,10 @@ ifeq ($(HAVE_CAP),y)
 	CFLAGS += $(HAVE_CAP_CFLAGS)
 	LDLIBS += $(HAVE_CAP_LDLIBS)
 endif
+ifeq ($(HAVE_RPC),y)
+	CFLAGS += $(HAVE_RPC_CFLAGS)
+	LDLIBS += $(HAVE_RPC_LDLIBS)
+endif
 
 # Rules can only be declared after all variables in them are known.
 %.o: %.c
diff --git a/configure b/configure
index 667dfbad..b0fad4b9 100755
--- a/configure
+++ b/configure
@@ -353,8 +353,8 @@ check_tirpc()
 		echo "HAVE_RPC:=y" >>$CONFIG
 		echo "yes"
 
-		echo 'LDLIBS +=' "$(${PKG_CONFIG} --libs libtirpc)" >>$CONFIG
-		echo 'CFLAGS += -DHAVE_RPC' "$(${PKG_CONFIG} --cflags libtirpc)" >>$CONFIG
+		echo 'HAVE_RPC_LDLIBS +=' "$(${PKG_CONFIG} --libs libtirpc)" >>$CONFIG
+		echo 'HAVE_RPC_CFLAGS += -DHAVE_RPC' "$(${PKG_CONFIG} --cflags libtirpc)" >>$CONFIG
 	else
 		echo "no"
 	fi
